### PR TITLE
Remove coord/vertex from API

### PR DESCRIPTION
**Pull request by Libbum**
Here's some work connected to #148. There was a lot of different discussions there, and some have already been fixed via the `Space` depreciation and #159.

This PR drops the public facing `coord2vertex` and `vertex2coord` function and replaces their use in example cases where needed. 

One place where this hasn't been done explicitly is `src/CA2D.jl`, which is going to be dropped in #128 anyhow.

Doesn't seem like this simplifies all that much, so would be good to get some feedback on what additional things could be done here.

### CODE REVIEW

**Comment by Libbum** [comment]
Additional:

Seems like `src/simulations/sample.jl` has a function thats never used: `clean_space!`

File: src/core/agent_space_interaction.jl [hunk 1]
```diff
@@ -13,11 +13,7 @@ move_agent_single!, kill_agent!, coord2vertex, vertex2coord, genocide!
 Remove an agent from model, and from the space if the model has a space.
 """
 function kill_agent!(agent::AbstractAgent, model::ABM{A, S}) where {A, S<:AbstractSpace}
-  if typeof(agent.pos) <: Tuple
-    agentnode = coord2vertex(agent.pos, model)
-  else
-    agentnode = agent.pos
-  end
+  agentnode = coord2vertex(agent.pos, model)
```
Comment (kavir1698) [inline src/core/agent_space_interaction.jl]: Agents on `GraphSpace`  still have integers for their positions, don't they? If so, this change will be wrong.

File: src/core/agent_space_interaction.jl [hunk 2]
```diff
@@ -69,26 +65,21 @@ in the model and remove it from the old position
...
   # remove agent from old position
-  if typeof(agent.pos) <: Tuple
-    oldnode = coord2vertex(agent.pos, model)
-    splice!(model.space.agent_positions[oldnode], findfirst(a->a==agent.id, model.space.agent_positions[oldnode]))
-    agent.pos = vertex2coord(pos, model)  # update agent position
-  else
-    splice!(model.space.agent_positions[agent.pos], findfirst(a->a==agent.id, model.space.agent_positions[agent.pos]))
-    agent.pos = pos
-  end
+  oldnode = coord2vertex(agent.pos, model)
```
Comment (kavir1698) [inline src/core/agent_space_interaction.jl]: Same here.

**Comment by Datseris** [comment]
Yes you can remove that. It is a remnant of `genocide!`

File: test/api_tests.jl [hunk 3]
```diff
@@ -38,17 +38,9 @@ model1 = ABM(Agent1, GridSpace((3,3)))
...
-@test length(pos1) == 1
-@test pos1[1] == 1
 
 move_agent!(agent, (2,2), model1)
-
 @test agent.pos == (2,2)
-pos1 = model1.space.agent_positions[coord2vertex((1,1), model1)]
-@test length(pos1) == 0
-pos2 = model1.space.agent_positions[coord2vertex((2,2), model1)]
-@test pos2[1] == 1
```
Comment (kavir1698) [inline test/api_tests.jl]: Shouldn't the tests be updated instead of removed?

Comment (Libbum) [reply test/api_tests.jl]: Yep, I killed that too early. Will post a fix soon.

File: src/core/agent_space_interaction.jl (hunk 1 above)
Comment (Libbum) [reply src/core/agent_space_interaction.jl]: This *may* be the case and if so, this was some overzealous cutting. I'm writing a test suite for these functions & will update the result whatever the outcome.

Comment (Libbum) [reply src/core/agent_space_interaction.jl]: This wasn't a problem here, but was indeed the case for `move_agent!`. Fixed with 
0cbc01f.

**Comment by Libbum** [comment]
- `clean_space!` is in fact used in `sample!`, so it should remain.
- The latest commit adds a number of checks for `kill_agent!` and `move_agent!` to make sure they function as expected.

**Comment by Datseris** [comment]
I am really sorry, did I sleep on the wheel on this? Is it just good to go?

**Comment by Libbum** [comment]
Was just waiting on a sign off from Ali. Otherwise it should be good, yeah,

**Pull request merged**